toonz/sources/toonzqt/hexcolornames.cpp

```diff
@@ -0,0 +1,953 @@
+  // after a focus-in, this likely will fall into a hidden feature thought.
+  bool ctrlDown = event->modifiers() & Qt::ControlModifier;
+  if (!m_editing && !ctrlDown) selectAll();
+  m_editing = true;
+}
+
+//-----------------------------------------------------------------------------
+
+void HexLineEdit::focusOutEvent(QFocusEvent *event) {
+  QLineEdit::focusOutEvent(event);
+  deselect();
+  m_editing = false;
+}
+
+//-----------------------------------------------------------------------------
+
+void HexLineEdit::showEvent(QShowEvent *event) {
+  QLineEdit::showEvent(event);
+  updateColor();
+}
+
+//-----------------------------------------------------------------------------
+
+QCompleter *HexLineEdit::getCompleter() {
+  QStringList autolist;
+
+  // Build words list from all color names tables
+  HexColorNames::iterator it;
+  for (it = HexColorNames::beginMain(); it != HexColorNames::endMain(); ++it) {
+    autolist.append(it.name());
+  }
+  for (it = HexColorNames::beginUser(); it != HexColorNames::endUser(); ++it) {
+    autolist.append(it.name());
+  }
+
+  QCompleter *completer = new QCompleter(autolist);
+  completer->setCaseSensitivity(Qt::CaseInsensitive);
+  return completer;
+}
+
+//-----------------------------------------------------------------------------
+
+void HexLineEdit::onAutoCompleteChanged(bool enable) {
+  if (m_completer) {
+    m_completer->deleteLater();
+    setCompleter(nullptr);
+    m_completer = nullptr;
+  }
+  if (enable) {
+    m_completer = getCompleter();
+    setCompleter(m_completer);
+  }
+}
+
+void HexLineEdit::onColorsChanged() { onAutoCompleteChanged(true); }
+
+//-----------------------------------------------------------------------------
+
+//*****************************************************************************
+//  Hex color names editor
+//*****************************************************************************
+
+HexColorNamesEditor::HexColorNamesEditor(QWidget *parent)
+    : DVGui::Dialog(parent, true, false, "HexColorNamesEditor")
+    , m_selectedItem(nullptr)
+    , m_newEntry(false) {
+  setWindowTitle(tr("Hex Color Names Editor"));
+  setModal(false); // user may want to access main style editor and palettes
+
+  QPushButton *okButton    = new QPushButton(tr("OK"), this);
+  QPushButton *applyButton = new QPushButton(tr("Apply"), this);
+  QPushButton *closeButton = new QPushButton(tr("Close"), this);
+
+  m_unsColorButton = new QPushButton(tr("Unselect"));
+  m_addColorButton = new QPushButton(tr("Add Color"));
+  m_delColorButton = new QPushButton(tr("Delete Color"));
+  m_hexLineEdit    = new HexLineEdit("", this);
+  m_hexLineEdit->setObjectName("HexLineEdit");
+  m_hexLineEdit->setFixedWidth(75);
+
+  // Main default color names
+  QGridLayout *mainLay = new QGridLayout();
+  QWidget *mainTab = new QWidget();
+  mainTab->setLayout(mainLay);
+
+  m_mainTreeWidget = new QTreeWidget();
```

Comment:
To get rid of the space left of the name in the `Name` column, set the following:

```
 m_mainTreeWidget->setRootIsDecorated(false);
```
Same for `m_userTreeWidget`

Reply:
@justburner Would you be able to make this simple change to fix the spacing?  Once it's fixed, I will merge this.

Reply:
Ops sorry... got distracted with other projects.
I'll fix it asap.